homeassistant/components/climate/ambiclimate.py

```diff
@@ -0,0 +1,271 @@
+        self.oauth = oauth
+        self.store = store
+
+    async def get(self, request):
+        """Receive authorization token."""
+        hass = request.app['hass']
+
+        token_info = await self.oauth.get_access_token(request.query['code'])
+        await self.store.async_save(token_info)
+
+        await async_setup_platform(hass, self.config, self.async_add_entities)
+
+
+class Ambiclimate(ClimateDevice):
+    """Representation of a Ambiclimate Thermostat device."""
+
+    def __init__(self, heater, store):
+        """Initialize the thermostat."""
+        self._heater = heater
+        self._data = dict()
+        self._store = store
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._heater.device_id
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._heater.name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        return self._data.get('target_temperature')
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._data.get('temperature')
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {ATTR_CURRENT_HUMIDITY: self._data.get('humidity')}
+
+    @property
+    def is_on(self):
+        """Return true if heater is on."""
+        return self._data.get('power', '').lower() == 'on'
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._heater.get_min_temp()
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._heater.get_max_temp()
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def current_operation(self):
+        """Return current operation."""
+        return STATE_HEAT if self.is_on else STATE_OFF
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        await self._heater.set_target_temperature(temperature)
+
+    async def async_turn_on(self):
+        """Turn device on."""
+        await self._heater.turn_on()
+
+    async def async_turn_off(self):
+        """Turn device off."""
+        await self._heater.turn_off()
+
+    async def async_update(self):
+        """Retrieve latest state."""
+        token_info = await self._heater.control.refresh_access_token()
```

Comment:
Similar to the earlier comment -- do we need to handle when the token couldn't be refreshed?